benchmark/README.md

```diff
@@ -0,0 +1,56 @@
+
+
+## install.py
+`python install.py` should handle
+- install dependencies
+- download & preprocess dataset (if applicable)
+
+Note:
+- `torch` should not be a dependency, it has to work with latest pytorch master.
+
+## hubconf.py
+`hubconf.py` should contain a `class Model` with the following structure:
+
+```
+class Model:
+    device = None
+    jit = False
+
+    def set_device(self, device):
+        self.device = device
+
+    def set_jit(self, jit: bool):
+        self.jit = jit
+
+    def get_module(self):
+        """ Returns `model`, `example_input`
+        `model` should be torchscript model if self.jit is True
+        Both `model` and `example_input` should be on `self.device` properly."""
+
+    def get_eval_function(self):
+        """ Returns function `f` that triggers model evaluation that finishes in 30s.
+        `f` can be called with f(model, example_input)"""
+
+    def get_training_loop(self):
+        """ Returns a function `f` that triggers model training that finishes in 30s.
+        `f` can be called with f(model, example_input)"""
+```
+To test locally before submitting a PR, add the following snippet to `hubconf.py` and run `python hubconf.py`.
+```
+if __name__ == '__main__':
+    m = Model()
+    m.set_device('cpu')
+    model, example_input = m.get_module()
+    m.get_training_loop()(model, example_input)
```

Comment:
Training loop should probably not take any arguments. Some programs (GANs) won't have a single model. Similarly for the training loop: some things won't train off of the same example inputs that are used by the actual nn.Module. get_training_loop should probably just take a `niterations` argument and then handle its own data loading.

Reply:
Does `niterations` here refer to number of steps or epochs, or simply just repeat n times? 
The reason I'm asking is not all user code organize functions nicely around step/epoch so I feel repeat n times might be easier?

Reply:
Btw if training_loop/eval_function no longer takes any args, shall we just run them when calling them instead of returning the function? 
I'm thinking sth like 
```
m.train(niter=10) # repeat train 10 times
# current
m.get_training_loop()(niter=10)
```

Reply:
I feel `niter` can be a simple thing to do on user end, maybe we could just leave it outside the API for now?